Parquet/ResultListener/ParquetFile.cs

```diff
@@ -28,16 +30,26 @@ internal sealed class ParquetFile : IDisposable
 
         public Schema Schema { get; }
 
-        internal ParquetFile(Schema schema, Stream writeStream, ParquetFileOptions? options = null)
+        internal ParquetFile(Type generatingTool, Schema schema, Stream writeStream, ParquetFileOptions? options = null)
```

Comment:
Adding this type here seems a bit strange to me. What is meant by "generatingTool"?

Reply:
It is the tool that created the parquet file. So we can add it to a custom header. This way we know excactly which tool and what version created a given parquet file.

Reply:
I mean, what is a tool? I think this is too specific. Maybe we should just have a dictionary with custom metadata that can be set?

Reply:
There is nothing saying more fields cant be added. I just think it is good to have some information about what created the parquet file. I could make it optional, but i think it is usefull information for trouble shooting in the future.

Reply:
We discussed this in person, and decided to make custom metadata available as an API on top of parquet data. This change has been made now.